\documentclass[a4paper,11pt]{article}
\usepackage{ms}
%\usepackage{amsmath,amsfonts,amssymb}
\usepackage{natbib}
%\usepackage{lineno}
\usepackage{graphicx}
\usepackage{caption}
\modulolinenumbers[5]
%\linenumbers

%\usepackage{color}
%\definecolor{navy}{rgb}{0,0,0.4}
%\usepackage[colorlinks,citecolor=navy,linkcolor=navy,urlcolor=navy]{hyperref}

\IfFileExists{MinionPro.sty}{
\RequirePackage[mathlf,minionint]{MinionPro}
\RequirePackage[protrusion=true,expansion=false,verbose=silent]{microtype}
}{
\RequirePackage[osf]{mathpazo}
\def\textssc{\textsc}
\RequirePackage[tracking=smallcaps,protrusion=true,expansion=false,verbose=silent]{microtype}
}

\pdfminorversion=3

\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

\title{Comparative analysis of principal components can be misleading}
\author{
Josef C. Uyeda, Daniel S. Caetano, and Matthew W. Pennell
}

\date{}
\affiliation{
Department of Biological Sciences \& Institute for Bioinformatics and Evolutionary Studies, University of Idaho, Moscow, ID 83844, U.S.A. 
}
\corresponding{josef.uyeda@gmail.com}

\runninghead{PCA in comparative analyses}
\keywords{Phylogenetic comparative methods, principal components analysis, Brownian motion, Ornstein-Uhlenbeck, Early Burst, multivariate evolution}



\begin{document}

\mstitlepage
\parindent=1.5em
\addtolength{\parskip}{.3em}
\vfill
 
\finaltrue
\newpage
\singlespacing

\noindent Quantitative geneticists long ago recognized the value of studying evolution in a multivariate framework \citep{Pearson1903}. Due to linkage, pleiotropy, correlated selection, and mutational covariance, the evolutionary response in any phenotypic trait can only be properly understood in the context of other traits \citep{Lande1979, LynchWalsh}. This is of course also well-appreciated by comparative biologists. However, unlike in quantitative genetics, most of the statistical and conceptual tools for analyzing phylogenetic comparative data \citep[reviewed in][]{PennellHarmon} are designed for analyzing a single trait \citep[but see,][for exceptions]{RevellHarmon2008, RevellPCCA, Hohenlohe2008, RevellCollar2009, Schmitz2011, Bartoszek2012, Adams2014b,Adams2014}. Indeed, even classical approaches for testing for correlated evolution between two traits \citep[e.g.,][]{Felsenstein1985, Grafen1989, HarveyPagel1991} are not actually multivariate as each trait is assumed to have evolved under a process that is independent of the state of the other \citep{HansenOrzack2005}. As a result of these limitations, researchers with multivariate data are often faced with a choice: analyze each trait as if they are independent, or else decompose the dataset into statistically independent sets of traits, such that each set can be analyzed with the univariate methods.

Principal components analysis (PCA) is the most common method for reducing the dimensionality of the dataset prior to analyzing the data using phylogenetic comparative methods. PCA is simply a projection of multivariate data onto a new coordinate system. The first PC axis is the eigenvector in the direction of greatest variance, the second PC axis, the second greatest variance, and so on. While PCA is simply another way of representing a dataset, whether or not one can draw meaningful inferences from the PC axes will depend on both the question and the structure of the data. Evolution introduces a particular kind of structure into comparative data: as a result of shared common ancestry, close relatives are likely to share many traits and trait combinations. Performing comparative analyses without considering the species' evolutionary relationships is anathema to most evolutionary biologists, but the influence of phylogeny on data transformations is less understood \citep{Revell2008, Polly2013}.

Standard PCA continues to be regularly used in comparative biology. Researchers fit models to PC scores computed from a variety of trait types including geometric morphometric landmarks \citep[e.g.,][]{Dornburg2011, Hunt2013}, measurements of multiple morphological traits \citep[e.g.,][]{Harmon2010, BergmannIrshick2012, Weir2012, Pienaar2013, Price2014}, and climatic variables \citep[e.g.,][]{KozakWiens2010, Schnitzler2012}. The papers we have cited here are simply examples selected from a substantial number where standard PCA was used.

The most common approach for incorporating the non-independence of species is to assume a phylogenetic model for the evolution of measured traits and use the expected covariance in the calculation of the PC axes and scores \citep[phylogenetic Principal Components Analysis, or pPCA;][]{Revell2008}. Revell's method, explained in detail below, assumes that the measured traits have evolved under a multivariate Brownian motion (BM) process of trait evolution. \citet{Revell2008} demonstrated that standard PCA produces eigenvalues and eigenvectors that are not phylogenetically independent. 

In this paper, we first extend the argument of \citet{Revell2008} and demonstrate how eigenvalues obtained from PCA systematically distort biological inference in a predictable manner. Performing phylogenetic comparative analyses on standard PC axes can be positively misleading. This point has been made in other fields that deal with autocorrelated data, such as population genetics \citep{Novembre}, ecology \citep{Podani2002}, climatology \citep{Richman1986} and paleobiology \citep{Bookstein2012}. However, the connection between these previous results and phylogenetic comparative data has not been widely appreciated and standard PCs continue to be regularly used in the field. We hope our paper helps change this practice.

Second, as stated above, \citet{Revell2008} assumed that the measured traits had evolved under a multivariate BM process. As the pPC scores are not phylogenetically independent \citep{Revell2008, Polly2013}, one must use comparative methods to analyze them which will in turn require selecting an evolutionary model for the scores. The choice of model for the traits and the pPC scores are separate steps in the analysis \citep{Revell2008}. Researchers must assume a model for the evolution of the traits in order to obtain the pPC scores and then perform model-based inference on these scores. This introduces some circularity into the analysis: it seems likely that the choice of a model for the evolution of the traits will influence the apparent macroevolutionary dynamics of the resulting pPC scores. To our knowledge this effect has not been previously explored. Here we analyze simulated data to investigate whether assuming a BM model for the traits introduces systematic biases in the pPC scores when the generating model is different. We then analyze two real comparative datasets to understand the implications of these results for the types of data that researchers actually have; the traits in these datasets have certainly not evolved by a strict BM process.

Last, we consider the interpretation of evolutionary models fit to pPC axes and discuss the conceptual and statistical advantages and disadvantages of using pPCA compared to alternative approaches for studying multivariate evolution in a phylogenetic comparative framework. We argue that the statistical advantages of using pPC axes comes at a substantial conceptual cost and that alternative techniques are likely to be much more informative for addressing many evolutionary questions.

\section{Methods}
\subsection{Overview of pPCA}
Before describing our analyses, we briefly review standard and phylogenetic PCA and highlight the differences between the two \citep[see][for a more detailed treatment]{Polly2013}. In conventional PCA, a $m \times m$ covariance matrix $\mathbf{R}$ is computed from a matrix of trait values $\mathbf{X}$ for the $n$ species and $m$ traits
\begin{equation}\label{eq:rpca}
\mathbf{R} = (n-1)^{-1}(\mathbf{X} - \mathbf{1}\boldsymbol{\mu}^\intercal)^\intercal (\mathbf{X} - \mathbf{1}\boldsymbol{\mu}^\intercal)
\end{equation}
where $\boldsymbol{\mu}$ is a vector containing the means of all $m$ traits and $\mathbf{1}$ is a column vector of ones. We note that in many applications $\mathbf{X}$ may not represent the raw trait values. In geometric morphometrics for example, size, translation and rotation will often be removed from $\mathbf{X}$ prior to computing $\mathbf{R}$ \citep{RohlfSlice, Bookstein1997}. The eigenvalues $\mathbf{D}$ and eigenvectors $\mathbf{V}$ of $\mathbf{R}$ are then obtained using singular-value decomposition $\mathbf{R}=\mathbf{V}\mathbf{D}\mathbf{V}^{-1}$ or some related technique. The scores $\mathbf{S}$, the trait values of the species along the PC axes, are computed as
\begin{equation}\label{eq:Spca}
\mathbf{S}=(\mathbf{X} - \mathbf{1}\boldsymbol{\mu}^\intercal)\mathbf{V}
\end{equation}

Phylogenetic PCA differs from this procedure in two important ways \citep{Revell2008,Polly2013} . First the covariance matrix is weighted by the inverse of the expected covariance of trait values between taxa under a given model $\mathbf{\Sigma}$. Under a BM model of trait evolution, $\mathbf{\Sigma}$ is simply proportional to the matrix representation of the phylogenetic tree $\mathbf{C}$, such that $\Sigma_{i,j}$ is the shared path length between lineages $i$ and $j$ \citep{Rohlf2001}. Since only relative branch lengths matter under a multivariate BM process, we can simply set $\mathbf{\Sigma}=\mathbf{C}$ without loss of generality, though we note that the absolute magnitude of the eigenvalues will depend on the scale of the branch lengths. Including the expected covariance between trait values essentially just accounts for the expected covariance of the phylogeny. Second, the space is centered on the ``phylogenetic means'' $\mathbf{a}$ of the traits rather than their arithmetic means, which can be computed following \citet{RevellHarmon2008}:
\begin{equation}\label{eq:phymean}
\mathbf{a}=[(\mathbf{1}^\intercal \mathbf{\Sigma}^{-1} \mathbf{1})^{-1} 
\mathbf{1}^\intercal \mathbf{\Sigma}^{-1} \mathbf{X}]^\intercal
\end{equation}
In pPCA, Equation \ref{eq:rpca} is therefore modified as
\begin{equation}\label{eq:rppca}
\mathbf{R} = (n-1)^{-1}(\mathbf{X} - \mathbf{1a}^\intercal)^\intercal \mathbf{\Sigma}^{-1} (\mathbf{X} - \mathbf{1a}^\intercal)
\end{equation}
Similarly, $\mathbf{S}$ can be calculated for pPCA using Equation \ref{eq:Spca} but substituting the phylogenetic means for the arithmetic means
\begin{equation}\label{eq:Sppca}
\mathbf{S}=(\mathbf{X} - \mathbf{1a}^\intercal)\mathbf{V}
\end{equation}
where again, $\mathbf{V}$ is a matrix containing the eigenvectors of $\mathbf{R}$, in this case obtained from Equation \ref{eq:rppca}.

The effect of weighting the covariance and centering the space using phylogeny has an important statistical consequence \citep{Revell2008, Polly2013}. In PCA, each PC score is independent of all other scores from the same PC axis and from scores on other axes. Due to the phylogenetic structure of the data, this property of independence does not hold when using pPCA. Therefore it is necessary to analyze pPC scores using phylogenetic comparative methods, just as one would for any other trait \citep{Revell2008, Polly2013}. 

\subsection{Effect of PCA on model selection under multivariate Brownian motion}
We simulated 100 replicate datasets under multivariate Brownian motion to evaluate the effect of using standard versus phylogenetic PCA to infer the mode of evolution. For each dataset, we used \texttt{TreeSim} \citep{treesim} to simulate a phylogeny of 50 terminal taxa under a pure-birth process and scaled each tree to unit height. We then simulated a 20-trait dataset under multivariate Brownian motion. When analyzing phylogenetic comparative data, $\mathbf{R}$ is estimated from the data; here, we set $\mathbf{R}$ to be a known quantity and use it to simulate phylogenetically structured trait data. For each simulation, we generated a positive definite covariance matrix for $\mathbf{R}$, by drawing eigenvalues from an exponential distribution with a rate $\lambda = \text{1/100}$ and randomly oriented orthogonal eigenvectors. We then used this matrix to generate a covariance matrix for the tip states 
$\mathbf{X}\sim \mathcal{N}(\mathbf{0}, \mathbf{R} \otimes \mathbf{C})$ where $\otimes$ is the Kronecker product.
For each of the 100 simulated datasets, we computed PC scores using both standard methods and pPCA \citep[using the \texttt{phytools} package;][]{phytools}. We used \texttt{phylolm} \citep{HoandAne2014} to fit models of trait evolution to the original data and to all PC scores obtained by both methods. Following \citet{Harmon2010}, we considered three models of trait evolution: 1) BM; 2) Ornstein-Uhlenbeck with a fixed root \citep[OU:][]{ Hansen1997}; and 3) Early Burst \citep[EB:][]{Blomberg2003, Harmon2010}. We then calculated the Akaike Information Criterion weights (AICw) for each model/transformation/trait combination.

To explore the effect of trait correlation on inference, we conducted an additional set of simulations where $\mathbf{R}$ was varied from the above simulations to result in more or less correlated sets of phenotypic traits. We drew eigenvalues $\mathbf{m}$ from an exponential distribution and scaled these so that the leading eigenvalue $m_{\text{1}}$ was equal to 1. We then exponentiated this vector across a sequence of exponents ranging for $\ll$1 to $\gg$1; this gave us a series of covariance matrices ranging from highly correlated ($m_{\text{1}} = \text{1}; m_{\text{2}}, \ldots, m_{\text{20}} \approx \text{0}$) to nearly independent ($\mathbf{m} \approx \textbf{1}$), respectively. We chose the series of exponents to obtain a regular sequence of $m_{\text{1}} / \sum_{i=\text{1}}^{\text{20}} m_i$ from 0.05 to 1. For each set of eigenvalues, we simulated 25 datasets and estimated the slope of the relationship between the absolute size of phylogenetically independent contrasts \citep{Felsenstein1985} and the height of the node at which they were calculated \citep[i.e., the ``node height test'' of][]{FreckletonHarvey2006}. Under OU models, this relationship is expected to be positive, while under EB models this relationship is negative. BM models are expected not to show correlation between contrasts and height of the nodes.

\subsection{Effect of using PCA when traits are not Brownian}
We then simulated datasets under alternative  models of trait evolution. Because of difficulties in efficiently simulating large multivariate datasets of covarying traits under OU or EB models, we instead simulated 20 independent traits under BM, OU and EB for 50 taxa trees (as above, but setting all eigenvalues equal to one another). Of course, this is certainly not represenative of the process that have shaped real multivariate data, but considering this simple case allowed us to investigate how misspecifying the model of trait evolution can impact analyses under the simplest scenario.

For the BM simulations, we set $\sigma^2=\text{1}$. For OU, we set $\sigma^2=\text{1}$ and $\alpha=\text{2}$, such that the phylogenetic half-life log(2)/$\alpha$ \citep{Hansen2008} was equal to $\sim$ 0.35 of the total tree depth. For EB, we again set $\sigma^2=\text{1}$ and set $r$, the exponential rate of deceleration, to be log(0.02). 

As above, we fit BM, OU and EB models to the original data, PC scores and pPC scores for each simulated dataset and estimated parameters and AICw. In addition to the model-fitting and comparison, for every transformation, we applied two common diagnostic tests for deviation from BM-like evolution to all trait/PC axes. First, we calculated the slope of the node height test as described in the preceding section. Second, we characterized the disparity through time \citep{Harmon2003} using the \texttt{geiger} package \citep{geiger2}. 

Finally, we examined the scenario in which a set of traits each follow a model of evolution with unique evolutionary parameters. In particular, we use the accelerating-decelerating (ACDC) model of \citet{Blomberg2003} to generate independent trait datasets. This model is a general case of the EB model which allows both accelerating or decelerating rates of phenotypic evolution. Accelerating rates of evolution result in identical likelihoods as the OU model (assuming the root state is at the optimal trait value and the tree is ultrametric), and thus are equivalent for our purposes. (We provide a proof to support this claim in the Appendix included in the Supplementary Material.) We simulated 100 datasets with 50 taxa and 20 traits. Trees were generated as in previous simulations. Each trait was simulated along the phylogeny with an exponential rate of change $r$ drawn from a normal distribution with mean 0 and standard deviation of 5. Values of $r$ above 0 correspond to accelerating evolutionary rates, while those below 0 correspond to decelerating, or Early-Burst models of evolution. For each dataset, we conducted both standard and phylogenetic PCA in which the traits are standardized to unit variance (i.e., using correlation matrices, which ensured traits across parameter values had equal expected variances). For each PC or pPC, we regressed the magnitude of the trait loadings against the trait's ACDC parameter value. We then visualized whether there were systematic trends in the relationship between the ACDC parameter value, and the weight given to a particular trait across PC axes. Such systematic trends would indicate that either PCA or pPCA ``sorts'' traits into PC axes according to the particular evolutionary model each trait follows.

\subsection{Empirical examples}
We analyzed two comparative datasets assembled from the literature, allowing us to investigate the effects of principal components analyses on realistically structured data. First, we analyzed phenotypic evolution across the family Felidae (cats) using measurements from two independent sources -- five cranial measurements from \cite{slater_2009} and body mass and skull width from \cite{sakamoto_2010}. For the analysis, we used the supertree compiled by \cite{Nyakatura_2012}. Second, we analyzed 23 morphometric traits in \textit{Anolis} lizards and phylogeny from \cite{Mahler2010}. In both datasets, all measurements were linear measurements on the logarithmic scale. We conducted standard and phylogenetic PCA and examined the effect of each on model-fitting, the slope of the node height test, and the average disparity through time. All simulations and analyses were conducting using R v3.0.2. Scripts to reproduce all analyses are available at \url{https://github.com/mwpennell/phyloPCA}.
  
\section{Results}
\subsection{Effect of PCA on model selection under multivariate Brownian motion}
Standard PCA introduces a systematic bias in the favored model across principal components. In our simulations where the traits evolved under a multivariate BM model, EB models had systematically elevated support as measured by Akaike weights for the first few components, for which it generally exceeded support for the BM model (Figure \ref{corbm}). Fitting models sequentially across PC axes 1-20 revealed a regular pattern of increasing support for BM models moving from the first toward the intermediate components, followed by increasing support for OU models among later components, which generally approached an AICw of 1. This regular pattern across trait axes was not present for either the original datasets, or for phylogenetic principal components, which found strong support for the BM model regardless of which trait was analyzed. As both BM is a special case of both OU and EB, the likelihoods for the more complex models will converge on that of BM when the true model is Brownian. AIC weights for model $i$ are computed as
$\text{AICw}_i = e^{\text{0.5}(\text{AIC}_{min} - \text{AIC}_i)}/\sum_j e^{\text{0.5}(\text{AIC}_{min} - \text{AIC}_j)}$
%\[\text{AICw}_i = \frac{e^{\text{0.5}(\text{AIC}_{min} - \text{AIC}_i)}}{\sum_j e^{\text{0.5}(\text{AIC}_{min} - \text{AIC}_j)}}\]
and therefore if the likelihoods are identical, OU and EB will have a $\Delta \text{AIC} = \text{AIC}_{min} - \text{AIC}_i = \text{2}$ (as OU and EB each have one more parameter than BM). The theoretical maximum for the AICw of BM is thus $1/(2e^{-1} + 1) \approx \text{0.576}$.
 
Multivariate datasets simulated with high correlations (low effective dimensionality) showed increased support for BM across PC axes. When the leading eigenvalue explained a large proportion of the variance, the slope of the node height test converged toward 0, indicating no systematic distortion of the contrasts through time (Figure \ref{rank}). However, when the eigenvalues of the rate matrix were more even, standard PCA resulted in a negative slope in the node height test among the first few PCs, which in turn provides elevated support for EB models. This pattern is reversed among higher PC axes, which have positive slopes between node height and absolute contrast size, which provides elevated support for OU models (Figures \ref{rank} and \ref{nhplot}). 

\subsection{Effect of using PCA when traits are not Brownian}
If the underlying model was either OU or EB rather than BM, then phylogenetic PCA tended to increase support for the true model relative to the original trait variables for the first few component axes (Figures \ref{oufit}, \ref{aicwbm}, and \ref{aicweb}). For example, when each of the original trait variables were simulated under an OU process, support for the OU model increased for pPC1 relative to the original trait variables. Higher principal component axes inferred a regular pattern of decreasing support for the OU model, while the last few PCs have equivocal support for either a BM or OU model (Figure \ref{oufit}). Furthermore, parameter estimation was affected by phylogenetic PCA. The $\alpha$ parameter of the OU model was estimated to be stronger than the value simulated for individual traits for the first few pPC scores and lower for the higher components (Figure \ref{alpha}). 

Examining the outcomes of the node height tests (Figure \ref{nhplot}) and the disparity through time analyses (Figure \ref{dttplot}) helps clarify the results we observed from model comparison and parameter estimation. Under OU models, traits are expected to have the highest contrasts near the tips, whereas under EB models, traits will have the highest contrasts near the root of the tree. Under multivariate BM, standard PCA maximizes the overall variance explained across the entire dataset, thereby tending to select linear combinations of traits that maximize the contrasts at the root of the tree. Thus, the first few PCs are skewed toward resembling EB models, while the last few PCs are skewed toward resembling OU models. By contrast, the effect of pPCA on the node height relationship depends on the generating model. When traits are evolved under an OU model, the first few pPC axes show an exaggerated pattern of high variance towards the tips. Likewise, when traits are evolved under an EB model, the first few pPC axes show an exaggerated pattern of high variance towards the root of the tree. For traits generated under both OU and EB models, the higher components resemble BM-like patterns. 

When the data includes traits evolved under different parameters, both PCA and pPCA resulted in the systematic assignment of traits with particular parameter values to each PCA axes. Traits which follow EB models are preferentially given higher loadings under both PCA and pPCA for the first few PCs as well as the last few PCs (Figure \ref{ACDC}). Furthermore, both PCA and pPCA assign fewer PCs with higher loadings to traits that follow EB models, whereas the intermediate PCs have more even loadings slightly skewed toward accelerating rates (i.e., OU-like models). This asymmetry may reflect the fact that EB models are more variable in their outcomes to the phylogeny, owing to the fewer independent branches among which divergence can occur closer to the root of the tree. Our results indicate that both pPCA and PCA can be biased in the selection of PC axes with respect to the generating evolutionary model.

\subsection{Empirical examples}
In the Felid dataset, the seven morphometric traits were extremely highly correlated, with the first PC explaining 96.9\% and 93.7\% of the total variation in the dataset for standard PCA and phylogenetic PCA, respectively. All raw traits and the first PC axis of both standard and phylogenetic PCA support a BM model of evolution (PC and pPC axes have AICw's of 0.574, which is near the theoretical maximum for BM). The last four standard PC axes show strong support for a OU model (AICw $\approx$ 1) whereas under phylogenetic PCA the last axes have mixed support favouring BM or OU (Figure \ref{felidae.aicw}). Both the node height test and the disparity through time plots show this same pattern. The node height slope of the first axis is approximately zero while the slope of the remaining axes are slightly positive under standard and phylogenetic PCA. The first axis show the same disparity through time pattern of the untransformed data in both standard and phylogenetic PCA. However, the last PC axes show disparity accumulated toward the tips under standard PC, while phylogenetic PCA produced a less clear pattern (Figure \ref{felidae.nh}).

For the morphometric traits in the \textit{Anolis} dataset, the first PC also explained a large proportion of the variation (92.6\% and 90.0\% for standard and phylogenetic PCA, respectively). Most of the untransformed traits had equivocal support for either a BM or EB model (Figure \ref{anoles_aicw}). While PC1 of both PCA and pPCA mirrored this pattern, the remaining PCs for both PCA and pPCA show a general pattern of decreasing support for an EB model (Figure \ref{anoles_aicw}). Collectively PCs 2-4 had higher support for the EB model than any other PC in both standard PCA ($\text{AICw}_{EB}$: PC2 $=$ 1.0; PC3 $=$ 0.47; PC4 $=$ 0.28) and phylogenetic PCA ($\text{AICw}_{EB}$: pPC2 $=$ 1.0; pPC3 $=$ 0.43, pPC4 $=$ 0.27).  Similarly, these early PC axes tended to have more negative slopes from the node height test relative to the average trait in the dataset (Figure \ref{anoles_nh}).

\section{Discussion}
Different ways of representing the same set of data can change the meaning of measurements and alter the interpretations of subsequent statistical analyses \citep{Houle2011}. PCA is often considered to be a simple linear transformation of a multivariate dataset and the potential consequences of performing phylogenetic comparative analyses on PC scores has received very little attention. In this paper, we sought to highlight the fact that fitting macroevolutionary models to a handful of PC axes may positively mislead inference -- what appears like the signal of an interesting biological process may simply be an artifact stemming from how PCA is computed. By focusing analyses exclusively on the first few PC axes, as is commonly done in comparative studies, researchers are, in effect, taking a biased sample of a multivariate distribution \citep{Mitteroecker}. We demonstrate how this biased sampling can affect inferences from both PCA and pPCA. In particular, we demonstrate that it can lead researchers to erroneously infer a pattern of decreasing rates of evolution through time in highly dimensional datasets.

We can obtain an intuitive understanding of how PCA can affect inferences by considering data simulated under a multivariate BM model. Despite a constant rate of evolution across each dimension of trait space, stochasticity will ensure that some dimensions will diverge more rapidly than expected early in the phylogeny, while others will diverge less. All else being equal, dimensions that happen to diverge early are expected to have the greatest variance across species and standard PCA will identify these axes as the primary axes of variation. However, the trait combinations that are most divergent early in the clade will appear to have slowed down towards the present simply due to regression toward the mean, resulting in the characteristic ``early burst'' pattern of evolution for the first few principal components. (\citealt{Pennell2012} pointed out that lineage diversification models are susceptible to a similar sampling effect.) An analogous process will result in the last few PCs following an OU process, in which the amount of divergence will appear to increase toward the present. Standard PCA thus effectively ``sorts'' orthogonal trait dimensions by whether they follow EB, BM and finally, OU like patterns of trait divergence. Traits studied using PCA may therefore often be biased to reflect particular evolutionary models, merely as a statistical artifact.

These problems ultimately stem from making statistical inferences from a selected few PC axes without accounting for how PCA transforms autocorrelated data. This issue is certainly not limited to phylogenetic comparative studies \citep[see][]{Richman1986, Podani2002, Novembre, Jolliffe2002, Bookstein2012}. For example, \citet{Novembre} demonstrated that apparent waves of human migration in Europe obtained from PCA of genetic data \citep[e.g.,][]{Cavalli} could be attributed to artifacts similar to those we document here; in their case, the auto-correlation was the result of geography rather than phylogeny. While the bias introduced by analyzing standard PCs with phylogenetic models has been documented previously \citep{Revell2008, Polly2013}, we sought to clarify precisely how inferences of macroevolutionary processes and patterns can be impacted.  

\citet{Revell2008} recognized the need for accounting for phylogenetic correlation when performing PCA transformations and introduced the phylogenetic PCA method. Our simulations verify that when the underlying model is multivariate BM, pPCA mitigates the effect of deep divergences among clades in the major axes of variation by scaling divergence by the expected divergence given the branch lengths of the phylogeny. However, BM is often a poor descriptor of the macroevolutionary dynamics of trait evolution \citep[for example, see][]{Harmon2010, Pennell-adequacy} and assuming this model when performing pPCA is less than ideal. \citet{Revell2008} suggested that alternative covariance structures could be used to estimate phylogenetically independent PCs for different models. For example, one could first optimize the $\lambda$ model \citep{Freckleton2002} across all traits simultaneously and then rescale the branch lengths of the tree according to the estimated parameter in order to obtain $\mathbf{\Sigma}$ for use in Equation \ref{eq:rppca}. However, one cannot compare model fits across alternative linear combinations of traits, so the data transformation must occur separately from modeling the evolution of the PC axes. As \citet{Revell2008} noted, parameters estimated to construct the covariance structure for the pPCA will likely be different from the same parameters estimated using the PC scores themselves. Furthermore, this procedure is restricted to models that assume a shared mean and variance structure across traits \citep[see][for examples where this does not apply]{Hansen2008, Hansen2012SysBio, Bartoszek2012}. As such, if the question of interest relies on model-based inference, transforming the data using pPCA necessitates ad-hoc assumptions about the evolution of the traits, and researchers must hope that the resulting inferences are generally robust to these decisions.

We show that when the trait model is misspecified, systematic and predictable distortions occur across pPC axes -- similar to those that were observed when the phylogeny was ignored altogether. In some scenarios such distortions may not substantially alter inference. For example, when all traits evolve under a OU model (or when all traits evolve under a EB model), we find that these distortions primarily serve to inflate the support for the true model. Even so, interpretation of parameter estimates for pPC scores becomes much more challenging (Figures \ref{nhplot}, \ref{dttplot}, and \ref{alpha}). More complex scenarios can produce more worrying distortions. When evolutionary rates vary through time and across traits, both PCA and pPCA will sort traits into PC axes according to which evolutionary model they follow, despite all traits being evolutionarily independent. Under the conditions we examined, this resulted in both PC1 and pPC1 being heavily-weighted toward EB-type models, despite simulating an even distribution of accelerating and decelerating rates across traits. We observe similar patterns for both PCA and pPCA in the \textit{Anolis} morphometric dataset (Figures \ref{anoles_aicw} and \ref{anoles_nh}). Focusing on the first few axes of variation identified by pPCA alone may skew our view of evolutionary processes in nature, and bias researchers toward finding particular patterns of evolution. 

When employed as a descriptive tool, PCA can be broadly used even when assumptions regarding statistical non-independence or multivariate normality are violated \citep{Jolliffe2002}. There is nothing wrong with using standard PCA or pPCA on comparative data to describe axes of maximal variation across species or for visualizing divergence across phylo-morphospace \citep{Sidlauskas2008}. Furthermore, our simulations and empirical analyses suggest that strong correlations among traits (i.e., when the leading eigenvector explained a majority of the variation, e.g., $>$ 75\%), PC scores may not be appreciably distorted (Figure \ref{rank}). The statistical artifacts we describe are more likely to appear when matrices have high effective dimensionality \citep[see][]{Bookstein2012}. Given that many morphometric datasets may be highly correlated, the overall effect of using standard PCA or of misspecifying the model in phylogenetic PCA may in some cases be relatively benign. 

And we certainly do not mean to imply that the biological inferences that have been made from analyzing standard or phylogenetic PC scores in a comparative framework are necessarily incorrect. When \citet{Harmon2010} analyzed the evolution of PC2 (what they referred to as ``shape'') obtained using standard PCA, they found very little support for the EB model across their 39 datasets. The magnitude of the bias introduced by using standard PCA is difficult to assess but any bias that did exist would be towards finding EB-like patterns. This only serves to strengthen their overall conclusion that such slowdowns are indeed rare \citep[but see][]{SlaterPennell}. Our results do suggest that in some cases analyses conducted with PC axes should be revisited to ensure that results are robust.

The broader question raised by our study is how one should draw evolutionary inferences from multivariate data. The first principal component axis from pPCA is the major axis of divergence across the sampled lineages in the clade (also known as the ``line of divergence''). This axis is of considerable interest in evolutionary biology. The direction of this line of divergence may be affected by the orientation of within-population additive genetic (co)variance $\mathbf{G}$, such that evolutionary trajectories may be biased along the ``genetic line of least resistance'' \citep[i.e., divergence occurs primarily along the leading eigenvector of $\mathbf{G}$, $\mathbf{g}_{\text{max}}$;][]{Schluter1996}. Alternatively, the line of divergence may align with $\boldsymbol{\omega}_{\text{max}}$, the  ``selective line of least resistance'', due to the structure of phenotypic adaptive landscapes \citep{Arnoldetal2001, Jonesetal2007, Arnoldetal2008}, or else may be driven by patterns of gene flow between populations \citep{Guillaume2007} or the pleiotropic effects of new mutations \citep{Jonesetal2007, Hether2013}. 

While it is perfectly sensible and interesting to compare the orientation of pPC1 to that of $\mathbf{g}_{\text{max}}$, $\boldsymbol{\omega}_{\text{max}}$ or other within-population parameters, making explicit connections between macro- and microevolution requires a truly multivariate approach. Quantitative genetic theory \citep[e.g.,][]{Lande1979, LynchWalsh} makes predictions about the overall pattern of evolution in multivariate space. By fitting evolutionary models to pPC scores, we are only considering evolution along these axes independently and are therefore missing much of what is going on. In contrast, 
multivariate tests for the correspondence of axes of trait variation within and between species can provide meaningful insights into the processes by which traits evolve over long time scales \citep{Hohenlohe2008, Bolstad2014}. 

The most conceptually straightforward multivariate approach for analyzing comparative data is to construct models in which there is a covariance in trait values between species (which is done in univariate models) and a covariance between different traits. Such multivariate extensions of common quantitative trait models have been developed \citep{ButlerKing2004, RevellHarmon2008, Hohenlohe2008, RevellCollar2009, motmot}. These allow researchers to investigate the connections between lines of divergence and within-population evolutionary parameters \citep{Hohenlohe2008} as well as to study how the correlation structure between traits itself changes across the phylogeny \citep{RevellCollar2009}. 

These approaches also have substantial drawbacks. First, the number of free parameters of the models rapidly increases as more traits are added \citep{RevellHarmon2008}, making them impractical for large multivariate datasets. This issue may be addressed by constraining the model in meaningful ways \citep{ButlerKing2004} or by assuming that all traits (or a set of traits) share the same covariance structure \citep{Klingenberg2013, Adams2014, Adams2014b}. Such restrictions of parameter space are especially appropriate for truly high-dimensional traits, such as shape. For such traits, we are primarily interested in the evolution of the aggregate trait and not necessarily the individual components \citep{Adams2014}. The second drawback is that these models allow for inference of the covariance between traits but the cause of this covariance is usually not tied to specific evolutionary processes. This difficulty can be addressed by explicitly modeling the evolution of some traits as a response to evolution of others. Hansen and colleagues have developed a number of models in which a predictor variable evolves via some process and a response variable tracks the evolution of the first as OU process \citep{Hansen2008, Hansen2012SysBio, Bartoszek2012}. This has been a particularly useful way of modeling the evolution of allometries \citep[e.g.,][]{Hansen2012SysBio, Voje2013, Bolstad2014}. But, as with the ``full covariance''  models discussed above, increasing the number of traits makes the model much more complex and parameter estimation difficult.

As we can only estimate a limited number of parameters from most comparative datasets -- and even when we consider large datasets, most existing comparative methods have only been developed for the univariate case -- it often remains necessary to reduce the dimensionality of a multivariate dataset to one or a few compound traits. We believe that although PCA can be potentially quite usefully applied to this problem, it may be in ways that are statistically and conceptually distinct from how it is conventionally applied to comparative data. 

First, we argue that reducing multivariate problems to more easily managed, lower dimensionality analyses should be approached with the specific goal of maintaining biological meaning and interpretability \citep{Houle2011}. The common practice of examining only the first few PCs carries with it the implicit assumption that PCA ranks traits by their evolutionary importance, though this is not necessarily true \citep{Polly2013}. If a certain PC axis is of sufficient biological interest in its own right, it may not matter if it is a biased subset of a multivariate distribution. The fact that a vast majority of the traits studied in adaptive radiations likely represent very biased axes of variation across the multivariate process of evolution does not diminish the importance of the inferences made from studying these traits. 

The danger occurs when the biological significance of the set of traits is poorly understood, and when the source of the statistical signal may be either artifactual or biological. If a trait was not of interest \textit{a priori}, then this essentially turns into a multiple comparisons problem in which PCA searches multivariate trait space for an unusual axis of variation. These axes will tend to appear to have evolved by a process inconsistent with the generating multivariate process as a whole. \textit{A posteriori} interpretation of the PC axes by their loadings is something of an art -- one must ``read the tea leaves'' to understand what these axes mean biologically. Even when a particular axis is correlated with a biological interpretation, it can be unclear whether the statistical signal supporting a particular inference results from the evolutionary dynamics of the trait of interest or if it is the result of statistical artifacts introduced by the imperfect representation of that trait by a PC axis. More rigorous algorithms can be applied to identify subsets of the original variables that best approximate the principal components, which although still biased, are frequently more interpretable \citep{Cadima2001,Somers1986, Somers1989, Hausman1982, Vines2000, Jolliffe2002, Zou2006}. Another potential approach is to use principal components computed from within-population data, rather than comparative data. For example, if $\mathbf{G}$, or failing that, the phenotypic variance-covariance matrix $\mathbf{P}$, is available for a focal species, then the traits associated to the principal axes of variation in that species can be measured across all species in the phylogeny. In other words, across species trait measurements can be projected along $\mathbf{g}_{\text{max}}$. This alleviates the issues we discuss in this paper by estimating PCs from within-population data that is independent from the comparative data used for model-based inference.  

Components defined by within-population variance structure or by approximating principal components with interpretable linear combinations will not explain as much variance across taxa as standard PCA and will not necessarily be statistically independent of one another. But the extra variance explained by the principal components of comparative data may in fact include a sizeable amount of stochastic noise, rather than interesting biological trait variation, as we have shown in our simulations. Furthermore, while the trait combinations (eigenvectors) identified by pPCA will be statistically orthogonal, this is only true in the particular snapshot captured by comparative data and does not imply that they are evolving independently. The distinction between statistical and evolutionary independence is crucial \citep{HansenHoule2008} but it is easy to conflate these concepts when the data has been abstracted from its original form.  We argue that the added intepretability of carefully chosen and biologically meaningful trait combinations far outweighs the cost of some trait correlations or explaining less-than-maximal variation.

\section{Concluding remarks}
In this note, we sought to clarify some statistical and conceptual issues regarding the use of principal components in comparative biology. We have shown that fitting evolutionary models to standard PC axes can be postively misleading. And despite the development of methods to correct for this, in our reading of the empirical literature, we have found this to be a common oversight. We have also demonstrated that misspecifying the model of trait evolution when conducting pPCA may influence the inferences we make from the pPC scores. We show that in some scenarios, pPCA may sort traits according to the particular evolutionary models they follow in a similar manner as standard PCA. Ignoring phylogeny altogether is, of course, a form of model misspecification. Consequently, we caution that the use of pPCA may bias inference toward identifying particular evolutionary patterns, which may or may not be representative of the true multivariate process shaping trait diversification as a whole. We hope that our paper provokes discussion about how we should go about analyzing multivariate comparative data. We certainly do not have the answers but argue there are some major theoretical limitations inherent in using PCA, phylogenetic or not, to study macroevolutionary patterns and processes.

\section{Acknowledgements}

We would like to thank our advisor, Luke Harmon, for encouraging us to pursue this project and for providing insightful comments on the work and manuscript. We thank Luke Mahler for providing data for the \textit{Anolis} empirical example. We thank Frank Andersen, Tanja Stadler, David Polly, and two anonymous reviewers for helpful comments on this paper. JCU was supported by NSF DEB 1208912 and DBI 0939454. DSC was supported by a fellowship from Coordena\c{c}\~{a}o de Aperfei\c{c}oamento de Pessoal de N\'{i}vel Superior (CAPES: 1093/12-6). MWP was supported by a NSERC postgraduate fellowship. 

\newpage
\bibliographystyle{sysbio}
\bibliography{phylopca.bib}

%% Figure 1.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{./fig/mv-bm-aic.pdf}
\caption{Distribution of support for BM, OU and EB models when the generating model is a correlated multivariate BM model. Support for models were transformed onto a linear scale by calculating an overall model support statistic: $AICw_{OU} - AICw_{EB}$. Thus high values support OU, low values support EB, and intermediate values near 0 indicate BM-like evolution. Models were fit to each replicated dataset for each of 20 different traits which were taken either from PC scores (blue line) or phylogenetic PC scores (green line). Shaded regions indicate the 25$^{th}$ and 75$^{th}$ quantiles of the model support statistic for  100 replicated datasets. The red line indicates the average model support statistic averaged over all 20 original trait variables. Note that EB models have higher Akaike weights for the first few PCs of standard PCA, and that later PCs subsequently favor BM and finally, OU models. No such bias is found across traits for either the original data or pPCA.}
\label{corbm}
\end{figure}

%% Figure 2.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{./fig/onion.pdf}
\caption{Effect of trait correlations on the slope of the node height test for PC scores (left) and pPC scores (right) under a multivariate BM model of evolution. The red line is the aggregated data for all 20 traits on the original (untransformed) scale. The intensity of the colors are proportional to the ranking of the PC or pPC axes, stronger lines represent the first axes. When the leading eigenvector explains very little variation in the data and the effective dimensionality is high, the slope of node height test increases from negative to positive across PC axes. This indicates that under standard PCA, PC1 has higher contrasts near the root of the tree, while later PCs have higher contrasts near the tips (resulting in the pattern of model support observed in Figure \ref{corbm}). As the amount of variance explained by the principal eigenvector increases, the slope of the node height test approaches 0. No such effect is found for phylogenetic PCA.}
\label{rank}
\end{figure}

%% Figure 3.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{./fig/uncor-ou-aic.pdf}
\caption{Distribution of support for BM, OU and EB models when the generating model is a uncorrelated multivariate OU model. Support for models were transformed into a linear scale by calculating an overall model support statistic: $AICw_{OU} - AICw_{EB}$. Thus high values support OU, low values support EB, and intermediate values near 0 indicate BM-like evolution. Models were fit to each replicated dataset for each of 20 different traits which were taken either from PC scores (blue line) or phylogenetic PC scores (green line). Shaded regions indicate the 25$^{th}$ and 75$^{th}$ quantiles of the model support statistic for 100 replicated datasets. The red line indicates the average model support statistic averaged over all 20 original trait variables. Note that standard PCA results in Akaike weights that are skewed toward EB for the first few PCs of standard PCA, while  and that later PCs subsequently favor OU models. By contrast, pPCA results in Akaike weights that are skewed toward stronger support for OU models relative to the original trait variables.}
\label{oufit}
\end{figure}

%% Figure 4.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{./fig/nh-2panel.pdf}
\caption{Relationship between the average phylogenetic independent contrasts and the height of the node across 100 datasets simulated under either a BM (left), OU (middle) or EB (right) model of evolution. Contrasts were calculated for each of the 20 traits corresponding to either PC scores (top row) or pPC scores (bottom row). Each line represents a best-fit linear model to the aggregated data across all 100 replicate simulations. Red lines are aggregated over all 20 traits on the original data. The plots are oriented so that the left side of each panel corresponds to the root of the phylogeny, with time increasing tipward to the right. The intensity of the colors are proportional to the ranking of the PC or pPC axes, stronger lines represent the first axes. PCA results in a predictable pattern of increasing slope in the contrasts across PCs. By contrast, pPCA only has systematic distortions across pPC axes when the underlying model is not multivariate BM. When this occurs, the first few pPC axes tend to have more extreme slopes than the original data (but in the correct direction).}
\label{nhplot}
\end{figure}

%% Figure 5.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{./fig/dtt-2panel.pdf}
\caption{Disparity through time plots averaged across the 100 simulated datasets. The datasets were simulated under BM (left), OU (middle) or EB (right). The analyses were then performed on PC scores (top row) and pPC scores (bottom row). The average disparity through time of all 20 original trait variables is indicated by the red line. We fit a loess curve through the relative disparities for each trait/transformation/model combination. The plots are oriented so that the left side of each panel corresponds to the root of the phylogeny, with time increasing tipward to the right. The intensity of the colors are proportional to the ranking of the PC or pPC axes, stronger lines represent the first axes. As in Fig. \ref{nhplot}, the first few axes from the PCA show a strong pattern of high disparity early in the clades' histories with the higher components showing seemingly higher disparity towards the present. Phylogenetic PCA corrects the distortion if the generating model is multivariate BM. However, if the generating model was not BM, the first few pPC axes tend to show an exaggerated pattern of disparity relative to the original traits.}
\label{dttplot}
\end{figure}

%% Figure 6.
\begin{figure}[p]
\centering
\includegraphics[scale=0.52]{./fig/acdc_slopes.pdf}
\caption{Relationship between factor loadings and ACDC parameter ($\alpha$) for PCA (left) and pPCA (right) across 100 simulated datasets. For each simulation a value of $\alpha$ were drawn from a Normal distribution with mean = 0 and sd = 5. Boxplots indicate the distribution of the slope of a linear model describing the relationship between the absolute factor loadings for a given PC and the magnitude of the ACDC parameter. A negative slope indicates that traits with decelerating rates of evolution tend to have higher loadings in that particular PC. Conversely, positive slopes indicate that traits with accelerating rates tend to have higher loadings.}
\label{ACDC}
\end{figure}

%% Figure 7.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{./fig/anoles_aicw.pdf}
\caption{Distribution of support for BM, OU and EB models for a 23-trait morphometric dataset taken from \cite{Mahler2010}. Support is measured in Akaike weights across all original trait variables (left), as well as standard PCA (middle) and pPCA (right). For both PCA and pPCA, support for the EB model appears to be concentrated in PCs 1-4, with a suggestive pattern of decreasing support across PCs 2-4. }
\label{anoles_aicw}
\end{figure}

\renewcommand\thefigure{S\arabic{figure}}
\renewcommand\thetable{S \arabic{table}}
\setcounter{figure}{0}    
\setcounter{table}{0}

%% Figure S1.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{./fig/uncor-bm-aic.pdf}
\caption{Distribution of support for BM, OU and EB models when the generating model is an uncorrelated multivariate BM model. Support for models were transformed into a linear scale by calculating an overall model support statistic: $AICw_{OU} - AICw_{EB}$. Thus high values support OU, low values support EB, and intermediate values near 0 indicate BM-like evolution. Models were fit to each replicated dataset for each of 20 different traits which were taken either from PC scores (blue line) or phylogenetic PC scores (green line). Shaded regions indicate the  25$^{th}$ and 75$^{th}$ quantiles of the model support statistic for  100 replicated datasets. The red  line indicates the average model support statistic averaged over all 20 original trait variables. Note that EB models have higher Akaike weights for the first few PCs of standard PCA, and that later PCs subsequently favor BM and finally, OU models. No such bias is found across traits for either the original data or pPCA axes.}
\label{aicwbm}
\end{figure}

%% Figure S2.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{./fig/uncor-eb-aic.pdf}
\caption{Distribution of support for BM, OU and EB models when the generating model is an uncorrelated multivariate EB model. Support for models were transformed into a linear scale by calculating an overall model support statistic: $AICw_{OU} - AICw_{EB}$. Thus high values support OU, low values support EB, and intermediate values near 0 indicate BM-like evolution. Models were fit to each replicated dataset for each of 20 different traits which were taken either from PC scores (blue line) or phylogenetic PC scores (green line). Shaded regions indicate the 25$^{th}$ and 75$^{th}$ quantiles of the model-support statistic for 100 replicated datasets. The red  line indicates the average model support statistic averaged over all 20 original trait variables. Note that both pPCA and PCA increase support for EB models for early PC axes.}
\label{aicweb}
\end{figure}

%% Figure S3.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{fig/alpha-est.pdf}
\caption{Estimated values of the $\alpha$ parameter from phylogenetic PCA when data is simulated under an uncorrelated multivariate OU model. The simulating value $\alpha=$ 2 is depicted with the red line. The estimate of $\alpha$ is inflated in the first few pPC axes consistent with an exaggerated support for the OU model. In the last pPC axes, $\alpha$ is estimated to be very close to 0, such that the OU model is statistically indistinguishable from a BM model. These results mirror those depicted in Figure \ref{oufit}.}
\label{alpha}
\end{figure}

%% Figure S4.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{fig/felidae_aicw.pdf}
\caption{Proportion of support for BM, OU and EB models for each of the traits/PC axes from the morphological dataset of Felidae species. Traits were log transformed prior to analysis. Note that all original traits and the first axes under standard and phylogenetic PCA show strong support for a BM model.}
\label{felidae.aicw}
\end{figure}

%% Figure S5.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{fig/felidae_nh-dtt.pdf}
\caption{Node height test and disparity through time plots for the morphological dataset of Felidae species. Each line represents a best-fit linear model (left) or loess curve fitted (right) to the original traits, PC or pPC scores. All traits were log transformed prior to analysis. The intensity of color is proportional to the ranking of the PC or pPC axes, stronger lines represent the first axes. Left panels show the relationship between the average phylogenetic independent contrasts and the height of the node. Red lines indicate the average value for the original trait values. Right panels show disparity through time plots. The plots are oriented so that the left side of each panel corresponds to the root of the phylogeny, with time increasing tipward to the right. Compare this highly correlated dataset with only 7 traits to the larger, less correlated dataset of \textit{Anolis} lizards (Figure \ref{anoles_nh}).}
\label{felidae.nh}
\end{figure}

%% Figure S6.
\begin{figure}[p]
\centering
\includegraphics[scale=0.65]{./fig/anoles_nh-dtt.pdf}
\caption{Node height test and disparity through time plots for the morphological dataset of \textit{Anolis} lizards. Each line represents a best-fit linear model (left) or loess curve fitted (right) to the original traits, PC or pPC scores. All traits were log transformed prior to analysis. The intensity of color is proportional to the ranking of the PC or pPC axes, stronger lines represent the first axes. Left panels show the relationship between the average phylogenetic independent contrasts and the height of the node. Red lines indicate the average value for the original trait values. Right panels show disparity through time plots. The plots are oriented so that the left side of each panel corresponds to the root of the phylogeny, with time increasing tipward to the right. }
\label{anoles_nh}
\end{figure}

\end{document}
