\documentclass[a4paper,12pt]{article}
\usepackage[osf]{mathpazo}
\usepackage{ms}
\usepackage{natbib}
\usepackage{lineno}
\usepackage{graphicx}
\usepackage{caption}
\modulolinenumbers[5]
\linenumbers

\pdfminorversion=3

\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

\title{A comment on the use of principal components in comparative biology}
\author{
Josef C. Uyeda$^{1,*}$, Daniel S. Caetano$^1$, and Matthew W. Pennell$^1$
}

\date{}
\affiliation{
 $^{1}$ Department of Biological Sciences \& Institute for Bioinformatics and Evolutionary Studies, University of Idaho, Moscow, ID 83844, U.S.A.\\ 
 $^{*}$ Email for correspondence: \texttt{pseudacris@gmail.com}\\
}

\runninghead{PCA in comparative analyses}
\keywords{Phylogenetic comparative methods, principal components, Brownian motion, Ornstein-Uhlenbeck, multivariate statistics}


\begin{document}

\mstitlepage
\parindent=1.5em
\addtolength{\parskip}{.3em}
\vfill

\section{Abstract}
\begin{enumerate}
\item A common procedure when analyzing multivariate data in a phylogenetic framework is to first transform the data into a set of principal component (PC) axes. Techniques for ``correcting for phylogeny'' when computing PCs have been developed and applied widely in the empirical literature. However, standard (i.e., uncorrected) principal components continue to be widely used in comprative analyses.

\item We demonstrate that failing to include phylogenetic relationships when calculating PC scores will mislead inferences in a predictable manner. Specifically, even when data is simulated under Brownian motion (BM), the first several PC scores will appear to have evolved rapidly early in a clade's history and slowed down towards the present day.

\item The primary method for calculating ``phylogenetically corrected'' PC scores relies on the assumption that the traits have evolved under a multivariate BM model. It is unknown whether this assumption is reasonable if the actual traits have evolved under more complex scenarios. Using simulations, we find that for many common models of evolution, calculating PC scores based on a BM model is a reasonable approximation though this approximation gets worse as the dimensionality of the analysis increases.

\item \emph{Synthesis:} While incorporating phylogeny when calculating PC scores does remove bias in many cases, a more fundamental issue remains when analyzing PCs in a comparative framework --- that drawing meaningful evolutionary inferences from models fit to PCs is challenging. Alternatives approaches for modeling multivariate data on a phylogeny are likely to be more informative, though further statistical and conceptual innovations will be necessary for these to be more broadly applicable.
\end{enumerate} 

\newpage

\section{Introduction}
The units of measurements are not neutral or arbitrary; different ways of representing the same set of data can change the meaning of the measurements and alter the interpretations of subsequent statistical analyses \citep{Hand2004, HansenHoule2008, Houle2011}. Therefore, careful attention needs to be paid to how data is measured and how it is transformed. Furthermore, this should be done with the scientific question in mind in order to ensure that resulting inferences are meaningful \citep{Houle2011}. 

Most approaches for studying evolution in a phylogenetic comparative framework were developed for univariate data \citep[reviewed in][]{PennellHarmon}. Even classical approaches for testing for correlated evolution across a phylogeny \citep[e.g.,][]{Felsenstein1985, Grafen1989, HarveyPagel1991} fundamentally model each trait as having evolved under an evolutionary process that is independent of the state of the other \citep{HansenOrzack2005}. However, the traits that we are actually interested in studying have likely evolved in concert with a suite of other traits and considering each trait is isolation may give a distorted picture of reality. Therefore, it is often desirable to reduce the multivariate dataset into a set of linear combinations of traits that are orthogonal to one another, such that each linear combination can then be analyzed with common comparative methods.

The most common method for reducing the dimensionality of the dataset is to perform Principal Components Analyses (PCA) prior to analyzing the data using phylogenetic comparative methods. The first PC axis is the eigenvector in the direction of greatest variance, the second PC axis, the second greatest variance, and so on. However, standard methods for calculating PC scores assume that the samples are independent of one another, which is hardly ever the case for comparative data --- as a result of shared common ancestry, relatives are likely to share many traits and trait combinations. This fact is of course, now almost universally recognized by biologists and conducting comparative analyses without considering the phylogenetic relationships of species is anathema to most evolutionary biologists. 

However, the necessity of considering phylogeny in some types of data transformations \citep{Revell2008} is not similarly recognized. Many researchers --- most of whom would never consider analyzing comparative data in a non--phylogenetic context --- continue the practice of conducting comparative analyses on standard PCs and thus inadvertantly introduce substantial bias into their analyses. Typical examples of traits where this is done include geometric morphometric landmarks \citep[e.g.,][]{Dornburg2011, Hunt2013}, measurements of multiple morphological traits \citep[e.g.,][]{Harmon2010, BergmannIrshick2012, Weir2012, Pienaar2013, Price2014}, and climatic variables \citep[e.g.,][]{KozakWiens2010, Schnitzler2012}. We stress that we are not tyring to shame these authors and empahsize that the papers that we have cited are simply examples (picked more--or--less at random) from a substantial number of papers where this non--intuitive mistake was made.

Fortunately, multiple potential solutions exists to address this problem.  One frequently used method is that of \citet{Revell2008}, who developed a procedure (explained in detail below) for converting multivariate data into ``phylogenetically corrected'' PC axes (hereafter, PPCA), by assuming that the original data evolved along the phylogeny using a multivariate Brownian motion (BM) model of evolution. In a brief simulation study, \citet{Revell2008} demonstrated that if the underlying model for the traits was indeed a multivariate BM model, performing standard PCA gave biased estimates of the eigenvalues, whereas PPCA removed this bias.

In this brief contribution we first, extend the argument of \citet{Revell2008} and demonstrate that not only are the eigenvalues obtained from PCA biased but that they are biased in a systematic and predictable way. Performing comparative analyses on standard PC axes positively misleads inference. This point has been made in other fields that deal with autocorrelation between observations, such as population genetics \citep{Novembre}, ecology \citep{Podani2002}, climatology \citep{Richman1986} and paleobiology \citep{Bookstein2012}. However, the connection between these previous results and phylogenetic comparative data has not been explicitly made and standard PCs continue to be widely used in the field. We hope that our paper helps change this practice.

Second, as stated above, \citet{Revell2008} assumed that the measured traits had evolved under a multivariate BM model. This has the potential to introduce an odd circularity into the analysis. If we assume that the underlying traits have evolved under BM and calculate a set of PPC axes based on this assumption, it seems reasonable to suppose that if the true model of trait evolution were different, the PPC scores would be inferred to have evolved in a more BM--like manner than the traits themselves. To our knowledge, the effect of this bias has not been explored. We perform a small simulation study to investigate this effect.

Last, we consider the interpretation of evolutionary models fit to PPC axes and discuss the conceptual and statistical advantages and disadvantages of using PPCA compared to alternative approaches for studying multivariate evolution in a phylogenetic comparative framework.

\section{Methods}
\subsection{\emph{Overview of PPCA}}
We briefly review the approach of \citet{Revell2008} for computing PPC scores for each species. If we assume that the traits have each evolved under a model of trait evolution, we can estimate an evolutionary variance--covariance (vcv) matrix between the $m$ traits that have been measured for $n$ species \citep{RevellHarmon2008, RevellCollar2009}:
\begin{equation}\label{eq:r}
\mathbf{R} = \frac{(\mathbf{X} - \mathbf{1a}^\prime ) \mathbf{\Sigma}^{-1} 
(\mathbf{X} - \mathbf{1a}^\prime )}{n-1}.
\end{equation}
Here $\mathbf{X}$ is the $n \times m$ matrix of observed data, $\mathbf{\Sigma}$ is a $n \times n$ matrix with elements $\Sigma_{ij}$ being the expected covariance between traits in lineages $i$ and $j$, and $\mathbf{a}$ is a vector of phylogenetic means for all $m$ traits ($\mathbf{a}=[(\mathbf{1}^\prime \mathbf{\Sigma}^{-1} \mathbf{1})^{-1} 
\mathbf{1}^\prime \mathbf{\Sigma}^{-1} \mathbf{X}]^\prime$). \emph{[include plain language description of what $\mathbf{R}$ actually represents.]}
Under a BM model of trait evolution, $\mathbf{\Sigma}$ simply mirrors the structure of the phylogeny, such that $\Sigma_{i,j}$ is the sum of shared branch lengths (from the root to the most recent common ancestor) between lineages $i$ and $j$. (We note that as only relative branch lengths matter for these calculations, we can arbitrarily set the BM rate parameter $\sigma^2$ equal to unity.) 

\emph{[Nice explanation of how eigenvalues $\mathbf{D}$ and eigenvectors $\mathbf{V}$ of the phylogenetic]}

We obtain PC scores for each species by projecting the PC axes into the original space
\begin{equation}
\mathbf{S} = (\mathbf{X} - \mathbf{1a}^\prime )\mathbf{V}
\end{equation}
and the loadings $L$ for each trait $i$ on the $j^{\mathrm{th}}$ PPC axis
\begin{equation}
L_{ij} = \frac{K_{ij}}{\sqrt{R_{ii}D_{jj}}}
\end{equation}
where $\mathbf{K}$ is the cross--covariance matrix $\mathbf{K}=(n-1)^{-1}(\mathbf{X}-\mathbf{1a}^\prime)^\prime \mathbf{\Sigma}^{-1} \mathbf{S}$ \citep[see][for details and explanation]{Revell2008}.

The end result of this procedure is a set of PPC axes, each representing of linear combinations of traits, and scores for all species for each PPC axis. The PPC scores are in the original species space and are therefore phylogenetically non--independent in the same way that the measured traits are.  


\subsection{\emph{Simulations}}

We did x

\subsection{\emph{Empirical examples}}

\section{Results}

\subsection{\emph{Simulations}}

\subsection{\emph{Empirical examples}}

\section{Discussion}
\subsection{\emph{Standard PCA should not be used for comparative analyses}} 

Discussion of main results (both simulated and empirical)

Tie into results from other studies (e.g., novembre)

These results certainly do not imply that the biological inferences made from analyzing standard PC scores in a comparative framework are necessarily incorrect.  
Interestingly, when \citet{Harmon2010} analyzed the evolution of PC2 (what they referred to as ``shape'') obtained using standard PCA, they found very little support for the ``Early Burst'' model \citep{Blomberg2003} across their 39 datasets. The fact that their use of standard PC axes biased their results \emph{towards an ``Early Burst'' pattern} only serves to strengthen their overall conclusion that such slowdowns are indeed rare \citep[but see][]{SlaterPennell}. However, our results do suggest that in some cases, analyses conducted with PC axes should perhaps be revisited to ensure that results are robust.

\subsection{\emph{Using PPCA when trait evolution is non-Brownian}}
In his original PPCA method, \citet{Revell2008} assumed that the original traits evolved under a multivariate BM model. He also suggested that one could potentially use alternative covariance structures. For example, one could first optimize the $\lambda$ model \citep{Pagel1999} across all traits simultaneously \citep[using the method of][]{Freckleton2002} and then rescale the branch lengths of the tree according to the estimated parameter in order to obtain $\mathbf{\Sigma}$ for use in Equation \ref{eq:r}. However, this is not a completely satisfying solution for two reasons. First, as noted by \citet{Revell2008}, the value of $\lambda$ used to construct the covariance structure for the PPCA will likely be different from the same parameter estimated using the PC scores themselves. Second, while this method could be extended to include other models \citep{ButlerKing2009, motmot}, this is restricted to models that assume a shared mean and variance structure across traits \citep[see][for examples where this does not  apply]{Hansen2008, Bartoszek2012}.

In our simulations, we found x.

This is because of y.

The effect of dimensionality

\subsection{\emph{The interpretation of PPCA and its alternatives}}

Aside from the statistical issues raised here, a broader, and more interesting, question is how one should draw evolutionary inferences from models of trait evolution fit to PPC axes. While PC (or PPC) axes are convenient representations of multivariate data in univariate space, this comes at the cost of abstracting the data that is analyzed from the questions that are being asked. 

First, this data transformation abstracts the analyses from the study of evolutionary processes. In quantitative genetics terminology, the first principal component axis of comparative data can be considered as the leading eigenvector of the phenotypic vcv matrix $\mathbf{P}$. If we make the (probably perilous) assumption that $\mathbf{P}$ is a proxy for the genetic vcv matrix $\mathbf{G}$, then to we must make the (extremely perilous) assumption that the vector of selection is perfectly aligned with PC1. As this assumption is extremely unlikely to be valid, fitting models to PC axes makes the models difficult (if not impossible) to interpret in terms of evolutionary processes. For some questions this abstraction may be acceptable. For instance, if the goal of fitting models is to make broad inferences about evolutionary patterns \citep{Freckleton2011, Hunt2012}, then we may not care if the models are connected to any process. In fact, many commonly used trait models \citep[notably, the ``Pagel'' tree transformations $\lambda, \delta, \kappa$][]{Pagel1997, Pagel1999} have no real bearing on any evolutionary processes \citep{HansenOrzack2005} and therefore using PCs in conjunction with these models therefore does not seem to come at a great cost. Other models, can be tied to specific evolutionary processes \citep{HansenMartins1996, EstesArnold2007, Hansen2008, Hansen2012SysBio, PennellHarmon, PennellPE} and using PC axes necessarily breaks this tie.

Second, and perhaps more importantly, we are interested in studying the evolution of phenotypes, not PC axes. Researchers often use broad terms to translate the PC axes back to trait space, such as stating that PC1 is size and PC2 is shape \citep[e.g.,][]{Harmon2010, Price2014}. However, this is a very imprecise way of quantifying biologically interesting variation. It is often difficult to understand precisely what we are explaining when we say that the evolution of PC2 is well described by, say, an OU process.

A number of alternatives to principal components have been developed for the analysis of multivariate data in a phylogenetic context. Here we think it useful to distinguish between the analysis of multivariate data (measuring body size and leg length, for instance) and truly multidimensional traits (such as ``shape'' obtained from the analyses of geometric morphometric landmarks), in which we are interested in understanding the evolution of the measurements in aggregate. Different conceptually and statistical approaches may be preferred for these two different cases. We note that there is, of course, a gray area in between these two; for example, ``climate niche'' may be considered a multidimensional trait but we may also be interested in whether the evolutionary change is a response to temperature, moisture, or seasonality.

For analyses of multivariate trait data, it is fairly straightforward to construct multivariate forms of commonly used trait models (e.g., BM, OU, etc.). In the univariate case, one must estimate a vcv matrix ($\mathbf{\Sigma}$ above) representing the expected variances and covariances between taxa for the trait being studied \citep[see][for details]{Harmon2010, Omeara2012}. In the multivariate case, $\mathbf{\Sigma}$ is the expected variances and covariance between all taxa and all traits \citep{RevellHarmon2008, Hohenlohe2008, RevellCollar2009, ButlerKing2009, motmot}. For the simple case of multivariate BM, fitting this model requires computing the Kronecker product of $\mathbf{R}$ (from Equation \ref{eq:r}) and $\mathbf{C}$, the patristic distance matrix derived from the phylogeny (that is, $\mathbf{\Sigma} = \mathbf{R} \otimes \mathbf{C}$). $\mathbf{\Sigma}$ is therefore a $nm \times nm$ matrix (where as above, $n$ is the number of species in the phylogeny and $m$ is the number of measured traits). While parameters may be constrained ,such as forcing several traits to share the same variance structure in an OU model, clearly the number of parameters increases very quickly as more traits are included.

A more biologically realistic alternative is to directly model the evolution of one trait as a response to the evolution of other traits (rather than just testing whether they are correlated). Hansen and colleagues have developed a number of models in which a predictor variable evolves via some process and a response variable tracks the evolution of the first as OU process \citep{Hansen2008, Labra2009, Hansen2012SysBio, Bartoszek2012}. This has been a particular useful way of modeling the evolution of allometries \citep{Hansen2012SysBio, Voje2013, Voje2013b}. However, like the correlational models discussed above, increasing the number of traits makes the model much more complex and difficult to fit.

An alternative to PC axes that has not been widely discussed is [Josef's soapbox]

Phylogenetic models for the evolution of multidimensional traits has recently been a very active area of research \citep{Bookstein2012, Klingenberg2013, Monteiro2013, Adams2014, Adams2014b}. More on this [Daniel?]

Representing multivariate data as PC axes is certainly appealing. We can use the wealth of comparative methods designed for univariate data on each PC axis. However, we argue that emerging, truly multivariate, methods will allow us to get a better grip on the processes that have shaped phenotypic evolution through deep time. Much work is yet to be done in this area before. To borrow from \citet{HansenHoule2008} (in a slightly different context), there will certainly be ``theoretical and statistical challenges, but these seem a small price to pay for meaning'' (p. 1213).

\section{Concluding remarks}
In this note, we sought to clarify exactly how failing to consider phylogeny when transforming data into PC axes can be positively misleading. As the axes of greatest variance will tend to mirror the deepest splits in the phylogeny, the PC scores will appear to have evolved rapidly early in the clade's history and slowed down towards the present day. This can lead researchers to inadvertanly attribute to biology what is merely a statistical artifact. And despite the development of methods to current for this, in our reading of the literature, we have found that failing to consider phylogeny when calculating PCs to be a common oversight. We hope our paper raises awareness of these issues and helps empiricists make informed decisions when analyzing comparative data. Last, we want to emphasize the point that different transformations of the same data lead to fundamentally different evolutionary inferences. We echo calls \citep{HansenHoule2008, Houle2011} that researchers should always think carefully about what measurements are used for analyses and do so with the end question in mind.

\newpage

\bibliographystyle{jecol}
\bibliography{phylopca.bib}
\end{document}
